src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -33,6 +35,12 @@ public final class PermissionEvaluationUtils {
             String.format(SERVICE_RESOURCE_FORMAT, SERVICE_PATTERN_STRING, SERVICE_RESOURCE_TYPE_PATTERN_STRING,
                     SERVICE_RESOURCE_NAME_PATTERN_STRING), Pattern.UNICODE_CHARACTER_CLASS);
 
+    private static final String POLICY_VARIABLE_PATTERN_STRING = "\\$\\{iot:(\\w*+.\\w*+.\\w*+)}";
```

Comment:
```suggestion
    private static final String POLICY_VARIABLE_PATTERN = "\\$\\{iot:(\\w*+.\\w*+.\\w*+)}";
```
style nit: string is redundant

Reply:
can we just match on `${iot:Connection.Thing.ThingName}` rather than matching on \w?  When we need to support other variables we can revisit the approach, can keep it very simple for now